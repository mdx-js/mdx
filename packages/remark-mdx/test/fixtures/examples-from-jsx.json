{
  "type": "root",
  "children": [
    {
      "type": "heading",
      "depth": 1,
      "children": [
        {
          "type": "text",
          "value": "Examples from JSX",
          "position": {
            "start": {
              "line": 1,
              "column": 3,
              "offset": 2
            },
            "end": {
              "line": 1,
              "column": 20,
              "offset": 19
            }
          }
        }
      ],
      "position": {
        "start": {
          "line": 1,
          "column": 1,
          "offset": 0
        },
        "end": {
          "line": 1,
          "column": 20,
          "offset": 19
        }
      }
    },
    {
      "type": "paragraph",
      "children": [
        {
          "type": "text",
          "value": "From ",
          "position": {
            "start": {
              "line": 3,
              "column": 1,
              "offset": 21
            },
            "end": {
              "line": 3,
              "column": 6,
              "offset": 26
            }
          }
        },
        {
          "type": "link",
          "title": null,
          "url": "https://facebook.github.io/jsx/",
          "children": [
            {
              "type": "text",
              "value": "jsx",
              "position": {
                "start": {
                  "line": 3,
                  "column": 7,
                  "offset": 27
                },
                "end": {
                  "line": 3,
                  "column": 10,
                  "offset": 30
                }
              }
            }
          ],
          "position": {
            "start": {
              "line": 3,
              "column": 6,
              "offset": 26
            },
            "end": {
              "line": 3,
              "column": 44,
              "offset": 64
            }
          }
        },
        {
          "type": "text",
          "value": ".",
          "position": {
            "start": {
              "line": 3,
              "column": 44,
              "offset": 64
            },
            "end": {
              "line": 3,
              "column": 45,
              "offset": 65
            }
          }
        }
      ],
      "position": {
        "start": {
          "line": 3,
          "column": 1,
          "offset": 21
        },
        "end": {
          "line": 3,
          "column": 45,
          "offset": 65
        }
      }
    },
    {
      "type": "mdxJsxFlowElement",
      "name": "Dropdown",
      "attributes": [],
      "children": [
        {
          "type": "paragraph",
          "children": [
            {
              "type": "text",
              "value": "A dropdown list",
              "position": {
                "start": {
                  "line": 6,
                  "column": 3,
                  "offset": 80
                },
                "end": {
                  "line": 6,
                  "column": 18,
                  "offset": 95
                }
              }
            }
          ],
          "position": {
            "start": {
              "line": 6,
              "column": 3,
              "offset": 80
            },
            "end": {
              "line": 6,
              "column": 18,
              "offset": 95
            }
          }
        },
        {
          "type": "mdxJsxFlowElement",
          "name": "Menu",
          "attributes": [],
          "children": [
            {
              "type": "paragraph",
              "children": [
                {
                  "type": "mdxJsxTextElement",
                  "name": "MenuItem",
                  "attributes": [],
                  "children": [
                    {
                      "type": "text",
                      "value": "Do Something",
                      "position": {
                        "start": {
                          "line": 8,
                          "column": 15,
                          "offset": 119
                        },
                        "end": {
                          "line": 8,
                          "column": 27,
                          "offset": 131
                        }
                      }
                    }
                  ],
                  "position": {
                    "start": {
                      "line": 8,
                      "column": 5,
                      "offset": 109
                    },
                    "end": {
                      "line": 8,
                      "column": 38,
                      "offset": 142
                    }
                  }
                },
                {
                  "type": "text",
                  "value": "\n",
                  "position": {
                    "start": {
                      "line": 8,
                      "column": 38,
                      "offset": 142
                    },
                    "end": {
                      "line": 9,
                      "column": 1,
                      "offset": 143
                    }
                  }
                },
                {
                  "type": "mdxJsxTextElement",
                  "name": "MenuItem",
                  "attributes": [],
                  "children": [
                    {
                      "type": "text",
                      "value": "Do Something Fun!",
                      "position": {
                        "start": {
                          "line": 9,
                          "column": 15,
                          "offset": 157
                        },
                        "end": {
                          "line": 9,
                          "column": 32,
                          "offset": 174
                        }
                      }
                    }
                  ],
                  "position": {
                    "start": {
                      "line": 9,
                      "column": 5,
                      "offset": 147
                    },
                    "end": {
                      "line": 9,
                      "column": 43,
                      "offset": 185
                    }
                  }
                },
                {
                  "type": "text",
                  "value": "\n",
                  "position": {
                    "start": {
                      "line": 9,
                      "column": 43,
                      "offset": 185
                    },
                    "end": {
                      "line": 10,
                      "column": 1,
                      "offset": 186
                    }
                  }
                },
                {
                  "type": "mdxJsxTextElement",
                  "name": "MenuItem",
                  "attributes": [],
                  "children": [
                    {
                      "type": "text",
                      "value": "Do Something Else",
                      "position": {
                        "start": {
                          "line": 10,
                          "column": 15,
                          "offset": 200
                        },
                        "end": {
                          "line": 10,
                          "column": 32,
                          "offset": 217
                        }
                      }
                    }
                  ],
                  "position": {
                    "start": {
                      "line": 10,
                      "column": 5,
                      "offset": 190
                    },
                    "end": {
                      "line": 10,
                      "column": 43,
                      "offset": 228
                    }
                  }
                }
              ],
              "position": {
                "start": {
                  "line": 8,
                  "column": 5,
                  "offset": 109
                },
                "end": {
                  "line": 10,
                  "column": 43,
                  "offset": 228
                }
              }
            }
          ],
          "position": {
            "start": {
              "line": 7,
              "column": 3,
              "offset": 98
            },
            "end": {
              "line": 11,
              "column": 10,
              "offset": 238
            }
          }
        }
      ],
      "position": {
        "start": {
          "line": 5,
          "column": 1,
          "offset": 67
        },
        "end": {
          "line": 12,
          "column": 12,
          "offset": 250
        }
      }
    },
    {
      "type": "mdxJsxFlowElement",
      "name": "Box",
      "attributes": [],
      "children": [
        {
          "type": "mdxFlowExpression",
          "value": "\n  shouldShowAnswer(user) ?\n  <Answer value={false}>no</Answer> :\n  <Box.Comment>\n     Text Content\n  </Box.Comment>\n",
          "position": {
            "start": {
              "line": 15,
              "column": 3,
              "offset": 260
            },
            "end": {
              "line": 21,
              "column": 4,
              "offset": 391
            }
          },
          "data": {
            "estree": {
              "type": "Program",
              "start": 261,
              "end": 390,
              "body": [
                {
                  "type": "ExpressionStatement",
                  "expression": {
                    "type": "ConditionalExpression",
                    "start": 266,
                    "end": 387,
                    "test": {
                      "type": "CallExpression",
                      "start": 266,
                      "end": 288,
                      "callee": {
                        "type": "Identifier",
                        "start": 266,
                        "end": 282,
                        "name": "shouldShowAnswer",
                        "loc": {
                          "start": {
                            "line": 15,
                            "column": 8
                          },
                          "end": {
                            "line": 15,
                            "column": 24
                          }
                        },
                        "range": [266, 282]
                      },
                      "arguments": [
                        {
                          "type": "Identifier",
                          "start": 283,
                          "end": 287,
                          "name": "user",
                          "loc": {
                            "start": {
                              "line": 15,
                              "column": 25
                            },
                            "end": {
                              "line": 15,
                              "column": 29
                            }
                          },
                          "range": [283, 287]
                        }
                      ],
                      "optional": false,
                      "loc": {
                        "start": {
                          "line": 15,
                          "column": 8
                        },
                        "end": {
                          "line": 15,
                          "column": 30
                        }
                      },
                      "range": [266, 288]
                    },
                    "consequent": {
                      "type": "JSXElement",
                      "start": 295,
                      "end": 328,
                      "openingElement": {
                        "type": "JSXOpeningElement",
                        "start": 295,
                        "end": 317,
                        "attributes": [
                          {
                            "type": "JSXAttribute",
                            "start": 303,
                            "end": 316,
                            "name": {
                              "type": "JSXIdentifier",
                              "start": 303,
                              "end": 308,
                              "name": "value",
                              "loc": {
                                "start": {
                                  "line": 15,
                                  "column": 45
                                },
                                "end": {
                                  "line": 15,
                                  "column": 50
                                }
                              },
                              "range": [303, 308]
                            },
                            "value": {
                              "type": "JSXExpressionContainer",
                              "start": 309,
                              "end": 316,
                              "expression": {
                                "type": "Literal",
                                "start": 310,
                                "end": 315,
                                "value": false,
                                "raw": "false",
                                "loc": {
                                  "start": {
                                    "line": 15,
                                    "column": 52
                                  },
                                  "end": {
                                    "line": 15,
                                    "column": 57
                                  }
                                },
                                "range": [310, 315]
                              },
                              "loc": {
                                "start": {
                                  "line": 15,
                                  "column": 51
                                },
                                "end": {
                                  "line": 15,
                                  "column": 58
                                }
                              },
                              "range": [309, 316]
                            },
                            "loc": {
                              "start": {
                                "line": 15,
                                "column": 45
                              },
                              "end": {
                                "line": 15,
                                "column": 58
                              }
                            },
                            "range": [303, 316]
                          }
                        ],
                        "name": {
                          "type": "JSXIdentifier",
                          "start": 296,
                          "end": 302,
                          "name": "Answer",
                          "loc": {
                            "start": {
                              "line": 15,
                              "column": 38
                            },
                            "end": {
                              "line": 15,
                              "column": 44
                            }
                          },
                          "range": [296, 302]
                        },
                        "selfClosing": false,
                        "loc": {
                          "start": {
                            "line": 15,
                            "column": 37
                          },
                          "end": {
                            "line": 15,
                            "column": 59
                          }
                        },
                        "range": [295, 317]
                      },
                      "closingElement": {
                        "type": "JSXClosingElement",
                        "start": 319,
                        "end": 328,
                        "name": {
                          "type": "JSXIdentifier",
                          "start": 321,
                          "end": 327,
                          "name": "Answer",
                          "loc": {
                            "start": {
                              "line": 15,
                              "column": 63
                            },
                            "end": {
                              "line": 15,
                              "column": 69
                            }
                          },
                          "range": [321, 327]
                        },
                        "loc": {
                          "start": {
                            "line": 15,
                            "column": 61
                          },
                          "end": {
                            "line": 15,
                            "column": 70
                          }
                        },
                        "range": [319, 328]
                      },
                      "children": [
                        {
                          "type": "JSXText",
                          "start": 317,
                          "end": 319,
                          "value": "no",
                          "raw": "no",
                          "loc": {
                            "start": {
                              "line": 15,
                              "column": 59
                            },
                            "end": {
                              "line": 15,
                              "column": 61
                            }
                          },
                          "range": [317, 319]
                        }
                      ],
                      "loc": {
                        "start": {
                          "line": 15,
                          "column": 37
                        },
                        "end": {
                          "line": 15,
                          "column": 70
                        }
                      },
                      "range": [295, 328]
                    },
                    "alternate": {
                      "type": "JSXElement",
                      "start": 335,
                      "end": 387,
                      "openingElement": {
                        "type": "JSXOpeningElement",
                        "start": 335,
                        "end": 348,
                        "attributes": [],
                        "name": {
                          "type": "JSXMemberExpression",
                          "start": 336,
                          "end": 347,
                          "object": {
                            "type": "JSXIdentifier",
                            "start": 336,
                            "end": 339,
                            "name": "Box",
                            "loc": {
                              "start": {
                                "line": 15,
                                "column": 78
                              },
                              "end": {
                                "line": 15,
                                "column": 81
                              }
                            },
                            "range": [336, 339]
                          },
                          "property": {
                            "type": "JSXIdentifier",
                            "start": 340,
                            "end": 347,
                            "name": "Comment",
                            "loc": {
                              "start": {
                                "line": 15,
                                "column": 82
                              },
                              "end": {
                                "line": 15,
                                "column": 89
                              }
                            },
                            "range": [340, 347]
                          },
                          "loc": {
                            "start": {
                              "line": 15,
                              "column": 78
                            },
                            "end": {
                              "line": 15,
                              "column": 89
                            }
                          },
                          "range": [336, 347]
                        },
                        "selfClosing": false,
                        "loc": {
                          "start": {
                            "line": 15,
                            "column": 77
                          },
                          "end": {
                            "line": 15,
                            "column": 90
                          }
                        },
                        "range": [335, 348]
                      },
                      "closingElement": {
                        "type": "JSXClosingElement",
                        "start": 373,
                        "end": 387,
                        "name": {
                          "type": "JSXMemberExpression",
                          "start": 375,
                          "end": 386,
                          "object": {
                            "type": "JSXIdentifier",
                            "start": 375,
                            "end": 378,
                            "name": "Box",
                            "loc": {
                              "start": {
                                "line": 15,
                                "column": 117
                              },
                              "end": {
                                "line": 15,
                                "column": 120
                              }
                            },
                            "range": [375, 378]
                          },
                          "property": {
                            "type": "JSXIdentifier",
                            "start": 379,
                            "end": 386,
                            "name": "Comment",
                            "loc": {
                              "start": {
                                "line": 15,
                                "column": 121
                              },
                              "end": {
                                "line": 15,
                                "column": 128
                              }
                            },
                            "range": [379, 386]
                          },
                          "loc": {
                            "start": {
                              "line": 15,
                              "column": 117
                            },
                            "end": {
                              "line": 15,
                              "column": 128
                            }
                          },
                          "range": [375, 386]
                        },
                        "loc": {
                          "start": {
                            "line": 15,
                            "column": 115
                          },
                          "end": {
                            "line": 15,
                            "column": 129
                          }
                        },
                        "range": [373, 387]
                      },
                      "children": [
                        {
                          "type": "JSXText",
                          "start": 348,
                          "end": 373,
                          "value": "\n       Text Content\n    ",
                          "raw": "\n       Text Content\n    ",
                          "loc": {
                            "start": {
                              "line": 15,
                              "column": 90
                            },
                            "end": {
                              "line": 15,
                              "column": 115
                            }
                          },
                          "range": [348, 373]
                        }
                      ],
                      "loc": {
                        "start": {
                          "line": 15,
                          "column": 77
                        },
                        "end": {
                          "line": 15,
                          "column": 129
                        }
                      },
                      "range": [335, 387]
                    },
                    "loc": {
                      "start": {
                        "line": 15,
                        "column": 8
                      },
                      "end": {
                        "line": 15,
                        "column": 129
                      }
                    },
                    "range": [266, 387]
                  },
                  "start": 261,
                  "end": 390,
                  "loc": {
                    "start": {
                      "line": 15,
                      "column": 3
                    },
                    "end": {
                      "line": 15,
                      "column": 132
                    }
                  },
                  "range": [261, 390]
                }
              ],
              "sourceType": "module",
              "comments": [],
              "loc": {
                "start": {
                  "line": 15,
                  "column": 3
                },
                "end": {
                  "line": 15,
                  "column": 132
                }
              },
              "range": [261, 390]
            }
          }
        }
      ],
      "position": {
        "start": {
          "line": 14,
          "column": 1,
          "offset": 252
        },
        "end": {
          "line": 22,
          "column": 7,
          "offset": 398
        }
      }
    }
  ],
  "position": {
    "start": {
      "line": 1,
      "column": 1,
      "offset": 0
    },
    "end": {
      "line": 23,
      "column": 1,
      "offset": 399
    }
  }
}
