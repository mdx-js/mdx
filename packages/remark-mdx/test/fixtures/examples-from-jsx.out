# Examples from JSX

From [jsx](https://facebook.github.io/jsx/).

<Dropdown>
  A dropdown list

  <Menu>
    <MenuItem>Do Something</MenuItem>
    <MenuItem>Do Something Fun!</MenuItem>
    <MenuItem>Do Something Else</MenuItem>
  </Menu>
</Dropdown>

<Box>
  {
    shouldShowAnswer(user) ?
    <Answer value={false}>no</Answer> :
    <Box.Comment>
       Text Content
    </Box.Comment>
  }
</Box>
